src/DateRange/DateRange.tsx

```diff
@@ -0,0 +1,291 @@
+		DateRangeValueTypes.NoFilter
+	);
+	const [isShowFilter, setShowFilter] = React.useState<boolean>(false);
+
+	const handleSelected = (currentValue: string) => (e: any) => {
+		setPrevValue(value);
+		setValue(currentValue);
+	};
+
+	const onCloseFilter = () => {
+		setShowFilter(false);
+		setValue(prevValue);
+	};
+
+	const handleToggleFilter = (currentValue: string) => () => {
+		setPrevValue(value);
+		setValue(currentValue);
+		setShowFilter(!isShowFilter);
+	};
+
+	const handleApplyFilter = () => {
+		setShowFilter(false);
+	};
+
+	const handleChangeDateFrom = (currentDateFrom: any) => {
+		setDateFrom(currentDateFrom);
+	};
+
+	const handleChangeDateTo = (currentDateTo: any) => {
+		setDateTo(currentDateTo);
+	};
+
+	const parseDateToString = (date: Date) => {
+		return `${formatValue(date.getDate())}.${formatValue(
+			date.getMonth() + 1
+		)}.${date.getFullYear()}`;
+	};
+
+	const handleChangeTimeFrom = (hours: number, minutes: number) => {
+		setTimeFrom({ hours, minutes });
+	};
+
+	const handleChangeTimeTo = (hours: number, minutes: number) => {
+		setTimeTo({ hours, minutes });
+	};
+
+	const validationDates = () => {
+		if (dateFrom > dateTo) {
+			return PropError.Date;
+		} else if (
+			timeFrom.hours > timeTo.hours ||
+			(timeFrom.hours === timeTo.hours &&
+				timeFrom.minutes > timeTo.minutes)
+		) {
+			return PropError.Time;
+		} else {
+			return false;
+		}
+	};
+
+	const InnerEditorComponent = () => (
+		<div className="kit-date-range__popup">
+			<div className="kit-date-range__popup-dates">
+				<span className="kit-date-range__popup-date-text">
+					{radioConcreteFromText}
+				</span>
+				<div className="kit-date-range__popup-fields">
+					<div className="kit-date-range__popup-date">
+						<DateField
+							error={validationDates() === PropError.Date}
+							defaultDate={dateFrom}
+							noShadow={false}
+							onChange={handleChangeDateFrom}
+						/>
+					</div>
+					<div className="kit-date-range__popup-date">
+						<TimeField
+							error={validationDates() === PropError.Time}
+							hours={timeFrom.hours}
+							minutes={timeFrom.minutes}
+							noShadow={false}
+							onChange={handleChangeTimeFrom}
+						/>
+					</div>
+				</div>
+			</div>
+			<div className="kit-date-range__popup-dates">
+				<span className="kit-date-range__popup-date-text">
+					{radioConcreteToText}
+				</span>
+				<div className="kit-date-range__popup-fields">
+					<div className="kit-date-range__popup-date">
+						<DateField
+							error={validationDates() === PropError.Date}
+							defaultDate={dateTo}
+							noShadow={false}
+							onChange={handleChangeDateTo}
+						/>
+					</div>
+					<div className="kit-date-range__popup-date">
+						<TimeField
+							error={validationDates() === PropError.Time}
+							hours={timeTo.hours}
+							minutes={timeTo.minutes}
+							noShadow={false}
+							onChange={handleChangeTimeTo}
+						/>
+					</div>
+				</div>
+			</div>
+			{!!validationDates() && (
```

Comment:
Можно было просто
```suggestion
			{validationDates() && (
```
Но вообще лучше всё-таки использовать явное сравнение без приведения к булеву или любым другим типам